File: pizza-store-ui/src/app/core/models/category.ts
export interface Category {
  id: number;
  name: string;
  code: string;
  imageUrl: string;
  description: string;
  isActive: boolean;
  isDeleted: boolean;
  createdAt: string;
}

export interface CategoryRequest {
  name: string;
  code: string;
  imageUrl: string;
  description: string;
  isActive: boolean;
}

File: pizza_store_api/src/main/java/fa/training/pizza_store_api/model/Category.java
package fa.training.pizza_store_api.model;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Data;

import java.time.LocalDateTime;

@Data
public class Category {
    private int id;
    private String name;
    private String code;
    private String imageUrl;
    private String description;
    @JsonProperty("isActive")
    private boolean isActive;
    @JsonProperty("isDeleted")
    private boolean isDeleted;
    private LocalDateTime createdAt;
}
